### PR TITLE
all times are now ISO 1806 ONLY WORKS ON NOT WINDOWS

### PR DESCRIPTION
**Pull request by Evantm**

### CODE REVIEW

**Comment by Evantm** [comment]
please test this on linux before merge

File: scraper.py [hunk 1]
```diff
@@ -82,14 +84,24 @@ def process_outage(outage_el, status_type):
...
         outage["status"] = status
 
     return outage
 
+def wilfred(time_str) -> str:
+    from datetime import datetime
+
+    time_str = time_str.replace('a.m.','AM')
+    time_str = time_str.replace('p.m.','PM')
+    time_formated = datetime.strptime(time_str,"%b %d %-I:%M %p")
```
Comment (marcolussetti) [inline scraper.py:100]:
```suggestion
    time_formatted = datetime.strptime(time_str,"%b %d %-I:%M %p")
```

File: scraper.py [hunk 2]
```diff
@@ -82,14 +84,24 @@ def process_outage(outage_el, status_type):
...
+    time_formated.replace(year=datetime.now().year)
```
Comment (marcolussetti) [inline scraper.py:101]:
```suggestion
    time_formatted.replace(year=datetime.now().year)
```

File: scraper.py [hunk 3]
```diff
@@ -82,14 +84,24 @@ def process_outage(outage_el, status_type):
     }
 
     if status_type == "restored" or status_type == "planned":
-        outage["end"] = status
+        wilfred(outage['end']) = status
```
Comment (marcolussetti) [inline scraper.py:87]:
```suggestion
        status = wilfred(outage["end"])
```

File: scraper.py [hunk 4]
```diff
@@ -82,14 +84,24 @@ def process_outage(outage_el, status_type):
...
     return outage
 
+def wilfred(time_str) -> str:
+    from datetime import datetime
+
+    time_str = time_str.replace('a.m.','AM')
+    time_str = time_str.replace('p.m.','PM')
+    time_formated = datetime.strptime(time_str,"%b %d %-I:%M %p")
+    time_formated.replace(year=datetime.now().year)
+    return time_formated.iso()
```
Comment (marcolussetti) [inline scraper.py:102]:
```suggestion
    return time_formatted.iso()
```

File: scraper.py [hunk 5]
```diff
@@ -116,6 +128,4 @@ def main():
     with open("bchydro_outages.json", "w") as outfile:
         json.dump(output, outfile, indent=2, sort_keys=True)
 
-
-if __name__ == "__main__":
-    main()
+main()
```
Comment (marcolussetti) [inline scraper.py:131]:
```suggestion
if __name__ == "__main__":
    main()
```

I mean it doesn't make much of a difference but I'd stick with the convention.

File: scraper.py [hunk 6]
```diff
@@ -82,14 +84,24 @@ def process_outage(outage_el, status_type):
...
+        wilfred(outage['end']) = status
         outage["start"] = off_since
     elif status_type == "current":
-        outage["start"] = off_since
+        wilfred(outage['start']) = off_since
         outage["status"] = status
 
     return outage
 
+def wilfred(time_str) -> str:
```
Comment (marcolussetti) [inline scraper.py:95]:
```suggestion
def parse_time(time_str) -> str:
```

Boring but I guess we should probably make it more readable if we can

File: scraper.py [hunk 7]
```diff
@@ -82,14 +84,24 @@ def process_outage(outage_el, status_type):
...
+    from datetime import datetime
```
Comment (marcolussetti) [inline scraper.py:96]: We should probable move this to be grouped with the other imports

File: scraper.py [hunk 8]
```diff
@@ -72,6 +72,8 @@ def process_outage(outage_el, status_type):
     except:
         last_updated = ""
 
+
```
Comment (marcolussetti) [inline scraper.py:75]: Should probably be removed as the double newline, pep8, etc.

**Review by marcolussetti — DISMISSED** [review]
Will run tomorrow morning to see if it parses but looks good besides the comments.

**Comment by marcolussetti** [comment]
I'm curious about it not working on Windows as you'd think that standard library should be predictable -- I'll try looking into it tomorrow

**Comment by marcolussetti** [comment]
So it turns out that `datetime.strptime` is perfectly happy to take `%I` and `%d` as non-zero leading or zero leading. So there is no need to use the `%-I` / `%#I` platform-sensitive format on parsing. On output, that's a different matter.

**Comment by marcolussetti** [comment]
👍

**Pull request merged**